File: python/dsflix_sdk/modules/news.py
from __future__ import annotations

from ..http import HttpClient


class NewsModule:
    def __init__(self, http: HttpClient) -> None:
        self.http = http

    def get_top(self, page: int | None = None, language: str | None = None, category: str | None = None) -> dict:
        return self.http.get("/api/v2/news/top", {"page": page, "language": language, "category": category})

    def search(self, query: str, page: int | None = None, language: str | None = None) -> dict:
        return self.http.get("/api/v2/news/search", {"query": query, "page": page, "language": language})

File: src/modules/news.ts
// ─────────────────────────────────────────────────────────────
//  DSFlix SDK  –  src/modules/news.ts
// ─────────────────────────────────────────────────────────────

import { HttpClient } from '../http';
import { NewsResponse, NewsTopOptions, NewsSearchOptions } from '../types';

export class NewsModule {
  constructor(private http: HttpClient) {}

  /** Get top entertainment/streaming headlines */
  getTop(options?: NewsTopOptions): Promise<NewsResponse> {
    return this.http.get('/api/v2/news/top', options as Record<string, string | number | boolean | undefined>);
  }

  /** Search for news articles by keyword */
  search(query: string, options?: NewsSearchOptions): Promise<NewsResponse> {
    return this.http.get('/api/v2/news/search', { query, ...options } as Record<string, string | number | boolean | undefined>);
  }
}
